### ISSUE TITLE
rspec-rails + rabl rendering views even if I don't want them to be rendered?

### ISSUE THREAD

**Issue by agibralter** [rspec/rspec-rails#565]
This is verbatim of my stackoverflow [question](http://stackoverflow.com/q/11125157/59639), but I did some debugging and think that it's not an issue with my setup.

I'm using rabl 0.6.13, rspec-rails 2.10.1, and rails 3.2.6.

I'm trying to spec my controllers in isolation, but for some reason my rabl templates are throwing me all sorts of undefined method exceptions on the mocks I use in my controller specs. I am not using render_views. I thought rspec did not process views unless you specify render_views in the controller spec. I've run debugger to ensure that render_views? evaluates to false in the before block rspec-rails inserts. Has anyone else run into this problem?

Basically I put a `debugger` in `Haml::Plugin.call` and saw that it was only breaking on that statement if my rspec context included `render_views` (using the param `format => :html`). Then I noticed that if I put that same debugger statement `ActionView::Template::Handlers::Rabl.call` the spec would break there regardless of whether or not I included `render_views` in the rspec context (using the param `format => :json`). Does that make sense? At first I thought it was rabl... but now it seems like something is going on in rails or rspec-rails that is causing rendering when the format is json or when the handler is rabl or some combination of both. Is that intended behavior?
